src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -47,6 +47,8 @@ internal sealed class PrivateImplementationDetails : DefaultTypeDef, Cci.INamesp
             new ConcurrentDictionary<ImmutableArray<byte>, MappedField>(ByteSequenceComparer.Instance);
 
         private ModuleVersionIdField _mvidField;
+        // Dictionary that maps from analysis kind to instrumentation payload field.
+        private readonly ConcurrentDictionary<int, InstrumentationPayloadField> _instrumentationPayloadFields = new ConcurrentDictionary<int, InstrumentationPayloadField>();
```

Comment:
I don't know. None of the other dictionaries in this class are being initialized lazily. I have no idea how much space an empty ConcurrentDictionary consumes vs. the extra code required to do a thread-safe lazy initialization.